### PR TITLE
Task Image uploading using Amazon S3 and Paperclip

### PR DESCRIPTION
**Pull request by aplazand20**
[Commands]

    docker-compose down
    docker-compose up --build -d
    docker-compose exec web rails db:migrate:reset

[Pre-conditions]
• Login as a user
• Click the “Become a host” button located on the header
• Fill in the forms and upload a image in the Photos member route

[Expected Output]
• User can upload an image or images
• User can delete an image or images
• Images can be found in an AWS S3 bucket under the name, "pokehost-shania"

[Notes]
• I will improve the UI once I’m done with all the functionalities
• Publish button cannot be of use as of now because I have to be able to upload an image first which when I can manage the listings
• I this is a new pull request for Image Uploading because there were conflicts with past commits. Sorry for the inconvinience.

### CODE REVIEW

File: app/views/photos/create.html.erb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<h1>Photos#create</h1>
```
Comment (charlotteds) [inline app/views/photos/create.html.erb]: Remove unnecessary files.

File: app/views/photos/destroy.html.erb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<h1>Photos#destroy</h1>
```
Comment (charlotteds) [inline app/views/photos/destroy.html.erb]: Remove unnecessary files.

File: db/migrate/20190819045650_create_active_storage_tables.active_storage.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# This migration comes from active_storage (originally 20170806125915)
```
Comment (charlotteds) [inline db/migrate/20190819045650_create_active_storage_tables.active_storage.rb]: What is this file for?

File: config/storage.yml [hunk 4]
```diff
@@ -7,12 +7,12 @@ local:
   root: <%= Rails.root.join("storage") %>
```
Comment (charlotteds) [inline config/storage.yml:7]: Why do you have to add configuration in this file?

File: app/controllers/photos_controller.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+require 'aws-sdk'
+
+class PhotosController < ApplicationController
+  def create
+    @property = Property.find_by(id: params[:property_id])
+    photos = params[:photos]
+
+    photos.each do |photo|
```
Comment (charlotteds) [inline app/controllers/photos_controller.rb:9]: Error when no photo is chosen and `Add Photos` button is clicked:
image

File: app/controllers/photos_controller.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+      @property.photos.create(image: photo)
+    end
+    redirect_to photos_property_url(@property), success: 'File successfully uploaded'
+  end
+
+  def destroy
+    Photo.find(params[:id]).destroy
```
Comment (charlotteds) [inline app/controllers/photos_controller.rb]: Add `ajax` to photo deletion. Page must not reload on every deletion.

File: app/controllers/properties_controller.rb [hunk 7]
```diff
@@ -18,15 +18,16 @@ def create
   end
 
   def edit
+    properties_param
```
Comment (charlotteds) [inline app/controllers/properties_controller.rb]: What is this line for?

File: app/controllers/properties_controller.rb [hunk 8]
```diff
@@ -18,15 +18,16 @@ def create
...
   end
 
   def update
-    if @property.update(property_params)
+    @property = Property.find(params[:id])
+    if @property.save
```
Comment (charlotteds) [inline app/controllers/properties_controller.rb]: Condition should contain property update rather than save.

**Review by charlotteds — COMMENTED** [review]
Notes:
- Fix layout for image upload:
image

File: config/storage.yml (hunk 4 above)
Comment (aplazand20) [reply config/storage.yml:7]: I was following a tutorial back when the AWS S3 didn't save the images. I will remove it.

**Review by charlotteds — APPROVED** [review]

**Pull request merged**